planner/core/logical_plan_builder.go

```diff
@@ -890,6 +894,11 @@ func (b *PlanBuilder) buildUnion(ctx context.Context, union *ast.UnionStmt) (Log
 
 	oldLen := unionPlan.Schema().Len()
 
+	for i := 0; i < len(union.SelectList.Selects); i++ {
+		b.handleHelper.popMap()
```

Comment:
If we pop all maps in this "clear handle node", then could it make these nodes not to push nil map?

Reply:
If we don't push `nil`.
We'll need this following codes I think:
```
func tail() {
	if tailPt == 0 {
		return nil
	}
	return stack[tailPt-1]
}
```
I think both of them is okay.

Reply:
Yeah. All right. But don't you think the second one is more clear?

Reply:
But union may be a subquery, the stack may not be nil before we enter the union.
Then don't push anything would be dangerous. We may pop wrong things out.

Reply:
That's fine...